collect_app/src/main/java/org/odk/collect/android/formmanagement/FormsUpdater.kt

```diff
@@ -0,0 +1,182 @@
+package org.odk.collect.android.formmanagement
+
+import android.content.Context
+import org.javarosa.core.reference.ReferenceManager
+import org.odk.collect.analytics.Analytics
+import org.odk.collect.android.R
+import org.odk.collect.android.analytics.AnalyticsUtils
+import org.odk.collect.android.formmanagement.matchexactly.ServerFormsSynchronizer
+import org.odk.collect.android.formmanagement.matchexactly.SyncStatusAppState
+import org.odk.collect.android.notifications.Notifier
+import org.odk.collect.android.preferences.keys.GeneralKeys
+import org.odk.collect.android.preferences.source.SettingsProvider
+import org.odk.collect.android.provider.FormsProviderAPI
+import org.odk.collect.android.storage.StoragePathProvider
+import org.odk.collect.android.storage.StorageSubdirectory
+import org.odk.collect.android.utilities.ChangeLockProvider
+import org.odk.collect.android.utilities.FormsDirDiskFormsSynchronizer
+import org.odk.collect.android.utilities.FormsRepositoryProvider
+import org.odk.collect.android.utilities.InstancesRepositoryProvider
+import org.odk.collect.android.utilities.TranslationHandler
+import org.odk.collect.forms.FormSourceException
+import java.io.File
+import java.util.stream.Collectors
+
+class FormsUpdater(
```

Comment:
Really like how this turned out and the naming seems so obvious now seeing it! 🎉